src/Symfony/Component/HttpFoundation/Tests/RequestTest.php

```diff
@@ -1427,6 +1427,16 @@ public function testGetAcceptableContentTypes()
         $this->assertEquals(array('application/vnd.wap.wmlscriptc', 'text/vnd.wap.wml', 'application/vnd.wap.xhtml+xml', 'application/xhtml+xml', 'text/html', 'multipart/mixed', '*/*'), $request->getAcceptableContentTypes());
     }
 
+    public function testGetAcceptableFormats()
+    {
+        $request = new Request();
+        $this->assertEquals(array(), $request->getAcceptableFormats());
+
+        $request = new Request();
+        $request->headers->set('Accept', 'text/html, application/xhtml+xml, application/xml;q=0.9, */*');
+        $this->assertEquals(array('html', 'xml'), $request->getAcceptableFormats());
```

Comment:
why would it only return html and xml if everything is acceptable according to `*/*`? I think this shows that the method is not clear and generic enough to be added to the core IMO.

Reply:
I can't see where I would possibly use this method. What would be useful is a similar method that accepts formats as argument and would return the ones that are acceptable according to the accept header. This would then allow to make use of `*/*` and this is what people usually need in REST APIs etc.